### DISCUSSION TITLE
'Generate Model' from source yml  feature in Dev containers

### DISCUSSION THREAD

**Discussion by DataGobes** [AltimateAI/vscode-dbt-power-user#218] [Q&A] [answered]
I have setup a linux dev container for vscode for our dbt projects. The dbt power user extension is completely working inside the container but one feature, which is the 'generate model' feature. Inside the yml file the 'Generate Model' option will not show up.

When I am outside of the container in the same repo on which the devcontainer is configured, it works without even having the python bridge loaded (no python interpreter selected) so I am thinking it does not have anything to do with the python setup inside the container.

However, I have no clue how to solve this, anybody might have a clue about this?

**Comment by mdesmet** [comment, marked as answer]
2 points:

- Without the python interpreter you wouldn't be able to generate a model as we depend on it to fetch the columns of the source relation
- The Generate model action should be rendered based on the yaml parsing. So not sure what's going on there.

The code is here:

https://github.com/innoverio/vscode-dbt-power-user/blob/e5319556d73574b4800ab5ee30bf1302df9ef5b9/src/code_lens_provider/sourceModelCreationCodeLensProvider.ts#L33

**Reply by DataGobes** [reply]
Thank you for responding. Yes, you are right of course in that Generate Model does not function when you click on it without python, but the 'Generate Model' link you click inside the source yaml will still show up.

Inside the container that link does not show up with the same source yaml file open.

Thanks for the link to the code, will see if that gets me anywhere. 👍

**Reply by mdesmet** [reply]
It should be very easy to run it in debug mode and understand what's going on on your project.

**Reply by DataGobes** [reply]
I have it figured out now. What I had done inside the dev container is associate .yml and .yaml files with the jinja-yaml filetype to highlight and lint jinja templating syntax inside the yamls. This is what broke the 'generate model' feature because when I remove the file associations from the devcontainer.json it started working.

**Reply by mdesmet** [reply]
Right that's still something i need to investigate.

**Reply by mdesmet** [reply]
Would you be willing to write a section on how to setup the devcontainer in the README?

**Reply by DataGobes** [reply]
Yes, would be happy to do that. Also willing to share my complete code so one can just put that in a .devcontainer folder in his project and off you go.

**Comment by anandgupta42** [comment]
@DataGobes some other users are interested in using the extension with Dev containers. It will benefit them if you can share the script and help write the section on dev-containers. We are looking forward to your contribution.

**Reply by DataGobes** [reply]
It took a while but I can share it if still wanted, I finally took the time to take out the sensitive parts.

**Reply by anandgupta42** [reply]
Hey @DataGobes - That would be awesome. Thanks a lot.
